File: samples/src/commonMain/kotlin/sample/_class/external/ThirdPartyClass.ts
import { sample } from "@kustom/Samples"

type Item = sample._class.external.js.Item
type ItemWithJsExport = sample._class.external.ItemWithJsExport

export default class ThirdPartyClass {
    constructor(public items: Item[]) {}
    getItemAt(at: number): Item {
        return this.items[at]
    }
    getItemFromJsExport(): ItemWithJsExport {
        return new sample._class.external.ItemWithJsExport(1337, "JsExport", 1234)
    }
}

File: samples/src/commonMain/kotlin/sample/_class/external/ThirdPartyClass.kt
@file:JsModule("./ThirdPartyClass")
@file:JsNonModule

package sample._class.external

external class ThirdPartyClass {
    val items: Array<Item> = definedExternally
    fun getItemAt(at: Int): Item = definedExternally
    fun getItemFromJsExport(): ItemWithJsExport = definedExternally
}
